### DISCUSSION TITLE
Swiper 的 onIndexChange 函数能不能 传递 Swiper.Item 的key 给我？

### DISCUSSION THREAD

**Discussion by shifenhutu** [ant-design/ant-design-mobile#5380] [Q&A] [answered]
### Version of antd-mobile

latest

### What is this feature about?

我需要准确知道，当前用户切换到 哪一个 item了 （用户究竟 当前 查看的哪个图 ）
每次切换，我只能通过 onIndexChange  获取到 index 这个无意义的索引

但是我在创建 Swiper.Item 的时候， 传入了一个key，这个key虽然只是react用来提升性能的， 但 如果 onIndexChange  除了返回索引， 还能把 这个key给我， 那我就可以用key快速知道 用户切换到了哪一个

这个key我就可以传入有意义的值， 如 userid， 如 图片id， 等等 

-----

我设想中是这样的：

```tsx
 <Swiper onIndexChange={(index,key)={ console.log(index, key) }}>
    <Swiper.Item key={"123"}>
         123的项目
      </Swiper.Item>
    <Swiper.Item key={"456"}>
         456的项目
      </Swiper.Item>
    <Swiper.Item key={"789"}>
         789的项目
      </Swiper.Item>
 </Swiper>
```

当我切换到  123的项目 ， 就会打印  0, 123  (这里的123 是 我传给 Swiper.Item 的key )

当我切换到  456的项目 ， 就会打印  1, 456 (这里的456 是 我传给 Swiper.Item 的key )

**Comment by GOWxx** [comment, marked as answer]
写了个 [demo](https://codesandbox.io/s/blue-moon-omut4u?file=/src/App.tsx)

用数组，索引就意义了，你的需求完全可以满足，想要其他的数据也可以在数组元素里加。

**Comment by miracles1919** [comment]
你的期望是通过 key 知道换到了哪一个，但实际上 key 不是必传的
```jsx
<Swiper>
  <Swiper.Item>1</Swiper.Item>
  <Swiper.Item>2</Swiper.Item>
</Swiper>
```
这种情况下返回 key 是没有意义的，index 反而是最有效的
